pkg/operator/configobservation/audit/observe_audit_policy.go

```diff
@@ -0,0 +1,74 @@
+package audit
+
+import (
+	gojson "encoding/json"
+	"fmt"
+
+	yaml2 "sigs.k8s.io/yaml"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+)
+
+const policyConfigMapName = "audit-policy"
+
+// ObserveAuditPolicy observes the openshift-config/audit-policy ConfigMap and merges it into the kube-apiserver config.
+func ObserveAuditPolicy(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	errs := []error{}
+	prevObservedConfig := map[string]interface{}{}
+
+	// copy non-empty .auditConfig.policyConfiguration from existingConfig to prevObservedConfig
+	policyPath := []string{"auditConfig", "policyConfiguration"}
+	existingPolicy, _, err := unstructured.NestedFieldNoCopy(existingConfig, policyPath...)
```

Comment:
don't we want to check if it is non-empty?

Reply:
We ignore nil, but leave everything else. I think this is correct. If somebody writes `{}` or `{garbage: 42}` we want the API server to interpret that, not us here.